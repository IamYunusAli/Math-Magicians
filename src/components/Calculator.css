.calculator {
  width: 400px;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.output {
  background-color: #a6a6ac;
  height: 20px;
  padding: 10px;
  display: flex;
  justify-content: flex-end;
}

.row {
  width: 100%;
  display: flex;
}

button {
  width: 100%;
  height: 100%;
  border: 0;
  padding: 10px;
  border-right: 1px solid #bbb;
  border-bottom: 1px solid #bbb;
}

.sign {
  background-color: #f39136ec;
}

.equal {
  width: 32%;
}
