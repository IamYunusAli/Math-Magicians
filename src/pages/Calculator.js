import React, { useState } from 'react';
import calculate from '../logic/calculate';
import styles from './Calculator.module.css';

const Calculator = () => {
  const [state, setState] = useState(0);

  const handleClick = (e) => {
    const num = e.target.innerText;
    setState((state) => calculate(state, num));
  };
  const { next, total } = state;
  const displayHandler = (e) => {
    const val = e.target.textContent;
    setState((state) => ({ ...state, total: val }));
  };
  return (
    <div className={styles.calculator}>
      <div className={styles.output} onChange={displayHandler}>
        {next || total || 0}
      </div>
      <div className={styles.row}>
        <button type="button" onClick={handleClick}>AC</button>
        <button type="button" onClick={handleClick}>+/-</button>
        <button type="button" onClick={handleClick}>%</button>
        <button className={styles.sign} type="button" onClick={handleClick}>รท</button>
      </div>
      <div className={styles.row}>
        <button type="button" onClick={handleClick}>7</button>
        <button type="button" onClick={handleClick}>8</button>
        <button type="button" onClick={handleClick}>9</button>
        <button className={styles.sign} type="button" onClick={handleClick}>x</button>
      </div>
      <div className={styles.row}>
        <button type="button" onClick={handleClick}>4</button>
        <button type="button" onClick={handleClick}>5</button>
        <button type="button" onClick={handleClick}>6</button>
        <button className={styles.sign} type="button" onClick={handleClick}>-</button>
      </div>
      <div className={styles.row}>
        <button type="button" onClick={handleClick}>1</button>
        <button type="button" onClick={handleClick}>2</button>
        <button type="button" onClick={handleClick}>3</button>
        <button className={styles.sign} type="button" onClick={handleClick}>+</button>
      </div>
      <div className={styles.row}>
        <button type="button" onClick={handleClick}>0</button>
        <button className={styles.equal} type="button" onClick={handleClick}>=</button>
      </div>
    </div>
  );
};

export default Calculator;
