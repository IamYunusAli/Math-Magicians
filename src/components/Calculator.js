import React, { PureComponent } from 'react';
import './Calculator.css';

class Calculator extends PureComponent {
  render() {
    return (
      <div className="calculator">
        <div className="output" />
        <div className="row">
          <button type="button">AC</button>
          <button type="button">+/-</button>
          <button type="button">%</button>
          <button className="sign" type="button">รท</button>
        </div>
        <div className="row">
          <button type="button">7</button>
          <button type="button">8</button>
          <button type="button">9</button>
          <button className="sign" type="button">x</button>
        </div>
        <div className="row">
          <button type="button">4</button>
          <button type="button">5</button>
          <button type="button">6</button>
          <button className="sign" type="button">-</button>
        </div>
        <div className="row">
          <button type="button">1</button>
          <button type="button">2</button>
          <button type="button">3</button>
          <button className="sign" type="button">+</button>
        </div>
        <div className="row">
          <button type="button">0</button>
          <button className="equal sign" type="button">=</button>
        </div>
      </div>
    );
  }
}

export default Calculator;
