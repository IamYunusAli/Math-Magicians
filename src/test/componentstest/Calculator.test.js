import renderer from 'react-test-renderer';
import Calculator from '../../pages/Calculator';

test('should match Caculator snapshot', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
