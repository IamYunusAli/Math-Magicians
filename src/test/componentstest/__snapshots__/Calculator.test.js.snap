// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Caculator snapshot 1`] = `
<div
  className="calculator"
>
  <div
    className="output"
    onChange={[Function]}
  >
    0
  </div>
  <div
    className="row"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="sign"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="row"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="sign"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="row"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="sign"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="row"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="sign"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="row"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="equal"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
